docs/REST.md

```diff
@@ -594,32 +593,30 @@ defined by each "real" type manually or plugged in by REST API trait companion.
 on `GenCodec`, which means that if `DefaultRestApiCompanion` is used then every type used in REST API trait
 that has a `GenCodec` instance will be serializable to `JsonValue`.
 
-### Single body serialization
+#### Custom body serialization
 
-Single body parameters (annotated as `@Body`) are serialized straight into `HttpBody`, which encapsulates
+Body parameters of [`@CustomBody`](#custombody) methods are serialized straight into `HttpBody`, which encapsulates
 not only raw content but also MIME type. This way you can define custom body serializations for your types and
-you are not limited to `application/json`.
+you are not limited to `application/json` and `application/x-www-form-urlencoded`.
 
 By default (if there are no more specific implicit instances defined),
 serialization to `HttpBody` falls back to `JsonValue` and simply wraps JSON string into HTTP body
 with `application/json` type. This means that all types serializable to `JsonValue` are automatically
-serializable to `HttpBody`.
+serializable as `HttpBody`.
 
 ### Result serialization
 
 Result type of every REST API method is wrapped into `Try` (in case the method throws an exception)
-and "serialized" into `Async[RestResponse]` where `Async` is a type alias defined in `io.udash.rest.raw.RawRest`
-package. `Async[T]` dealiases to `(Try[T] => Unit) => Unit` which means that it is a consumer of a callback of possibly
-failed computation of value of type `T`. It's the most raw, low-level representation of repeatable asynchronous
-computations.
+and translated into `Async[RestResponse]`. This means that the macro engine looks for an implicit instance of 
+`AsRaw[Async[RestResponse], Try[R]]` and `AsReal[Async[RestResponse], Try[R]]` for every HTTP method with result type `R`.
 
-This means that macro engine looks for an implicit instance of `AsRaw/AsReal[Async[RestResponse], Try[R]]`
-for every HTTP method with result type `R`.
+* `Async` is a type alias defined in `RawRest` object.
+  This is just a very raw, low-level representation of asynchronous computations which may be invoked many times. 
+  It serves as a common denominator for all possible asynchronous abstractions like `Future`, Monix `Task`, etc.
```

Comment:
The abstractions are not asynchronous themselves.

Reply:
These abstractions _do_ capture asynchronicity, don't they?

Reply:
I wouldn't define asynchronous as "captures asynchronicity", but it's just a nitpick ;)

Reply:
Actually I'm curious to hear how you would word it.

Reply:
I'd probably go with `for abstractions over asynchronicity, like` to avoid repetition on `computation`.